src/app/examples/form/form.effects.spec.ts

```diff
@@ -0,0 +1,63 @@
+import { TestBed } from '@angular/core/testing';
+import { provideMockActions } from '@ngrx/effects/testing';
+import { Action } from '@ngrx/store';
+import { EffectsMetadata, getEffectsMetadata } from '@ngrx/effects';
+import { Observable } from 'rxjs';
+import { LocalStorageService } from '@app/core';
+import { FormEffects, FORM_KEY } from './form.effects';
+import { Form } from './form.model';
+import { ActionFormUpdate } from './form.actions';
+import { cold } from 'jasmine-marbles';
+
+describe('FormEffects', () => {
+  let actions$: Observable<Action>;
+  let formEffect: FormEffects;
+  let metadata: EffectsMetadata<FormEffects>;
+  let localStorageService: any;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [
+        FormEffects,
+        provideMockActions(() => actions$),
+        {
+          provide: LocalStorageService,
+          useValue: jasmine.createSpyObj('LocalStorageService', ['setItem'])
+        }
+      ]
+    });
+    localStorageService = TestBed.get(LocalStorageService);
+    formEffect = TestBed.get(FormEffects);
+  });
+
+  it('should be created', () => {
+    expect(formEffect).toBeTruthy();
+  });
+
+  it('should not dispatch any action', () => {
+    metadata = getEffectsMetadata(formEffect);
+    expect(metadata.persistForm).toEqual({ dispatch: false });
```

Comment:
Since we're talking about `TestBed`, using `beforeEach` instead of functions, ...

I personally would want to remove all of the `getEffectsMetadata` tests.
I don't think they add much value and are made with a "I have to write tests, so I'll write tests" mindset.

Thoughts?

Reply:
@timdeschryver hmm, I can see your point. On the other hand, we are testing for the cases when `@Effect` is supposed to dispatch some action so I guess it's also makes sense to "enforce" that it doesn't. Without such a test change in implementation can cause weird business errors when `@Effect` is still dispatching some action which is not supposed to which leads to unexpected state changes?

Reply:
Won't that be tested with `expect(foo).toBeObservable(bar)`?